plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -284,6 +287,217 @@ public void testSelectivelyOptimizingLeavesEqualityDeletes()
+        }
+        icebergTable.newRowDelta().addDeletes(writer.toDeleteFile()).commit();
+
+        assertQuery("SELECT * FROM " + tableName, "SELECT * FROM nation WHERE NOT ((regionkey = 1 AND name = 'BRAZIL') OR regionkey = 2 OR name = 'ALGERIA')");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testMultipleEqualityDeletesNestedFields()
+            throws Exception
+    {
+        String tableName = "test_equality_delets_different_schemas_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " ( id BIGINT, root ROW(nested BIGINT, nested_other BIGINT))");
+        assertUpdate("INSERT INTO " + tableName + " VALUES (1, row(10, 100))", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES (2, row(20, 200))", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES (2, row(20, 200))", 1);
+        Table icebergTable = loadTable(tableName);
+        assertThat(icebergTable.currentSnapshot().summary().get("total-equality-deletes")).isEqualTo("0");
+        Path metadataDir = new Path(metastoreDir.toURI());
+        TrinoFileSystem fs = HDFS_FILE_SYSTEM_FACTORY.create(SESSION);
+
+        List<String> deleteFileColumns = ImmutableList.of("root.nested");
+        Schema deleteRowSchema = icebergTable.schema().select(deleteFileColumns);
+        List<Integer> equalityFieldIds = deleteFileColumns.stream()
+                .map(name -> deleteRowSchema.findField(name).fieldId())
+                .collect(toImmutableList());
+        for (int i = 1; i < 3; i++) {
+            String deleteFile = "delete_file_" + UUID.randomUUID();
+            Parquet.DeleteWriteBuilder writerBuilder = Parquet.writeDeletes(new ForwardingFileIo(fs).newOutputFile(new Path(metadataDir, deleteFile).toString()))
+                    .forTable(icebergTable)
+                    .rowSchema(deleteRowSchema)
+                    .createWriterFunc(GenericParquetWriter::buildWriter)
+                    .equalityFieldIds(equalityFieldIds)
+                    .overwrite();
+            EqualityDeleteWriter<Record> writer = writerBuilder.buildEqualityWriter();
+
+            Types.StructType nestedStructType = (Types.StructType) deleteRowSchema.findField("root").type();
+            Record dataDelete = GenericRecord.create(deleteRowSchema);
+            try (Closeable ignored = writer) {
+                Record nestedStruct = GenericRecord.create(nestedStructType);
+                nestedStruct.setField("nested", 20L);
+                writer.write(dataDelete.copy(ImmutableMap.of("root", nestedStruct)));
+            }
+            icebergTable.newRowDelta().addDeletes(writer.toDeleteFile()).commit();
+        }
+
+        // TODO: support read equality deletes with nested fields(https://github.com/trinodb/trino/issues/18625)
+        assertThatThrownBy(() -> query("SELECT * FROM " + tableName)).hasMessage("Multiple entries with same key: 3=root.nested and 3=nested");
```

Comment:
```suggestion
assertThatThrownBy(() -> query("SELECT * FROM " + tableName)).hasMessageContaining("Multiple entries with same key");
```

Reply:
You are right. I test on 421 version, this message is diffrent.